ambry-messageformat/src/test/java/com.github.ambry.messageformat/BlobPropertiesTest.java

```diff
@@ -33,23 +55,52 @@ public void basicTest() {
     final String contentType = "ContentType";
     final int timeToLiveInSeconds = 144;
 
-    BlobProperties blobProperties = new BlobProperties(blobSize, serviceId);
+    short accountId = BlobProperties.ACCOUNTID_CONTAINERID_DEFAULT_VALUE;
+    short containerId = BlobProperties.ACCOUNTID_CONTAINERID_DEFAULT_VALUE;
+    short issuerAccountId = BlobProperties.ACCOUNTID_CONTAINERID_DEFAULT_VALUE;
+    if (version == BlobPropertiesSerDe.Version2) {
+      accountId = Utils.getRandomShort(TestUtils.RANDOM);
+      containerId = Utils.getRandomShort(TestUtils.RANDOM);
+      issuerAccountId = Utils.getRandomShort(TestUtils.RANDOM);
+    }
+
+    BlobProperties blobProperties;
+    if (version == BlobPropertiesSerDe.Version1) {
```

Comment:
yeah I did give it a try. But I also need the args passed in to construct the BlobProps for verification purposes. Then, it boils down to two over loaded functions which I felt doesn't make much of a diff. Let me know your opinion.